Assets/Scripts/MovePlayer.cs

```diff
@@ -0,0 +1,237 @@
+    /// </summary>
+    private enum directions { Down, Up, Left, Right};
+
+    /// <summary>
+    /// The animator attached to the player
+    /// </summary>
+    private Animator playerAnimator;
+    /// <summary>
+    /// Is the player allowed to move? Will be false if paused, in a cutscene, etc.
+    /// </summary>
+    private bool canMove = true;
+    /// <summary>
+    /// Has the player received input to move?
+    /// </summary>
+    private bool hasReceivedMovementInput;
+    /// <summary>
+    /// Is the player in slow terrain?
+    /// </summary>
+    private bool isInSlowTerrain;
+    /// <summary>
+    /// Is the movement coroutine currently running?
+    /// </summary>
+    private bool isMoving;
+    /// <summary>
+    /// Stores movement input
+    /// </summary>
+    private Vector2 movementInput;
+    /// <summary>
+    /// Movement input normalized for easier use
+    /// </summary>
+    private Vector2 normalizedMovementDirection;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        playerAnimator = GetComponent<Animator>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        GetInput();
+    }
+
+    private void FixedUpdate()
+    {
+        if (canMove && hasReceivedMovementInput && !isMoving)
+            StartCoroutine(MovementRoutine());
+    }
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        ToggleSlowMovement(collision);
+    }
+
+    /// <summary>
+    /// Toggles slow movement when crossing the border of slow terrain
+    /// </summary>
+    /// <param name="collision">The trigger collider the player crossed</param>
+    private void ToggleSlowMovement(Collider2D collision)
+    {
+        if (collision.gameObject.tag == slowTerrainTag)
+            isInSlowTerrain = !isInSlowTerrain;
+    }
+
+    /// <summary>
+    /// Gets player input
+    /// </summary>
+    private void GetInput()
+    {
+        movementInput = new Vector2(Input.GetAxis("Horizontal"), Input.GetAxis("Vertical"));
+        if (movementInput.magnitude > 0)
+            hasReceivedMovementInput = true;
+        else
+            hasReceivedMovementInput = false;
+    }
+
+    /// <summary>
+    /// Adjusts the player to fit exactly into a tile
+    /// </summary>
+    private void SnapToGrid()
+    {
+        int positionX = Mathf.RoundToInt(transform.position.x);
+        int positionY = Mathf.RoundToInt(transform.position.y);
+        transform.position = new Vector3(positionX, positionY);
+    }
+
+    /// <summary>
+    /// Triggers the player's animations based on movement direction
+    /// </summary>
+    /// <param name="movementDirection">What direction is the player moving/facing?</param>
+    /// <param name="slowTerrain">Is the player in slow terrain?</param>
+    /// <param name="cannotMove">Is the player able to move?</param>
+    private void PlayAnim(directions movementDirection, bool slowTerrain, bool cannotMove)
```

Comment:
Rename "cannotMove" param to something about blocked by collision (to distinguish from other canMove variable)